.rhdh/scripts/install-rhdh-catalog-source.sh

```diff
@@ -166,8 +166,21 @@ spec:
     source: registry-proxy.engineering.redhat.com
 " > "$TMPDIR/ImageContentSourcePolicy_${ICSP_URL_PRE}.yml" && oc apply -f "$TMPDIR/ImageContentSourcePolicy_${ICSP_URL_PRE}.yml"
 
-echo "[INFO] Using iib from image $UPSTREAM_IIB"
-IIB_IMAGE="${UPSTREAM_IIB}"
+# check if the IIB we're going to install as a catalog source exists before trying to install it
+if [[ ! $(command -v skopeo) ]]; then
+  errorf "Please install skopeo 1.11+"
+  exit 1
+fi
+
+UPSTREAM_IIB_MANIFEST="$(skopeo inspect docker://${UPSTREAM_IIB} --raw || exit 2)"
+# echo "Got: $UPSTREAM_IIB_MANIFEST"
+if [[ $UPSTREAM_IIB_MANIFEST == *"Error parsing image name "* ]] || [[ $UPSTREAM_IIB_MANIFEST == *"manifest unknown"* ]]; then
+  echo "$UPSTREAM_IIB_MANIFEST"; exit 3
```

Comment:
nit: this check is done after creating the ICSP resources in the cluster:

```
$ .rhdh/scripts/install-rhdh-catalog-source.sh --latest --install-operator rhdh
[INFO] Adding ISCP to resolve references to images not on quay.io as if from quay.io
imagecontentsourcepolicy.operator.openshift.io/quay-io created
FATA[0001] Error parsing image name "docker://quay.io/rhdh/iib:latest-v4.14-x86_64xxx": reading manifest latest-v4.14-x86_64xxx in quay.io/rhdh/iib: manifest unknown 
```

Could this validation be done before instead?

Reply:
yep!